ui/src/main/resources/PhenoTips/SimilarCases.xml

```diff
@@ -1090,23 +1144,93 @@
   });
 
   var init = function() {
-    var maxResults = $('similarity_results__max_results');
-    var minScore = $('similarity_results__min_score');
-    $$('.similarity-results-container').each(function (container) {
-      var serviceInfo = container.down('input[name="service"]');
-      var mmeServer = container.down('input[name="mmeServer"]');
-      var omimSection = container.down('input[name="omimSection"]');
-      var serviceDisclaimer = container.down('input[name="serviceDisclaimer"]');
-      new PhenoTips.widgets.Matcher(new XWiki.Document().page, container, {
-        enableRefresh: true,
-        serviceURL: serviceInfo &amp;&amp; serviceInfo.value,
-        maxResults: maxResults &amp;&amp; maxResults.value,
-        minScore: minScore &amp;&amp; minScore.value,
-        mmeServer: mmeServer &amp;&amp; mmeServer.value,
-        omimSection: omimSection &amp;&amp; omimSection.value,
-        serviceDisclaimer: serviceDisclaimer &amp;&amp; serviceDisclaimer.value
+    // Quick hack to confirm we're on a patient page and widgets.Matcher is not initialized yet:
+    if (!$('patient-record-actions') || $('similar-cases-tab')) {
+      return;
+    }
+
+    var onSimilarCasesPage = $('similar-cases-container');
+
+    if (!onSimilarCasesPage && $('patient-consents-json')) {
+      // Check all required conscents:
+      // if not all required conscents are granted - do not generate Similar Cases tab
+      var consents = [];
+      try {
+        consents = JSON.parse($('patient-consents-json').value);
+      } catch(err) {
+        console &amp;&amp; console.log(err);
+      }
+      var requiredNotGranted = consents.filter( function(consent) { return consent.isRequired &amp;&amp; consent.status == "no"; } );
+      if (requiredNotGranted.length &gt; 0) {
+        return;
+      }
+    }
+
+    var _generateSimilarCasesTab = function (isActive) {
+      return _generateTab(
+        'similar-cases-tab', // element id
+        isActive,
+        'Similar Cases', //displayed text
```

Comment:
Would be nice if this was translatable. You could use the[ `phenotips.similarCases.title`](https://github.com/phenotips/patient-network/blob/39217da5d46ea45c6917c0052ecdee9e30e52b7e/similarity-data-impl/src/main/resources/ApplicationResources.properties#L1) key that was used for the heading I suggested to remove on line 60. I would also [change the translation for this key](https://github.com/phenotips/patient-network/blob/39217da5d46ea45c6917c0052ecdee9e30e52b7e/similarity-data-impl/src/main/resources/ApplicationResources.properties#L1) to be "Matches" or "Matches for this patient" or "Matches for <patient id> (to be discussed with others) so that it's more consistent with the terminology we use on the My Matches page.